sqlserver/datadog_checks/sqlserver/connection.py

```diff
@@ -560,6 +560,9 @@ def test_network_connectivity(self):
         host, port = split_sqlserver_host_port(self.instance.get('host'))
         if port is None:
             port = 1433
+            provided_port = self.instance.get('port')
+            if provided_port is not None:
+                port = provided_port
```

Comment:
this isn't the only place where this needs to be set, we need to update the value for the connection strings too if they port is set manually 

https://github.com/DataDog/integrations-core/blob/adb425873a6f59e9998a93b92a869acd88792c58/sqlserver/datadog_checks/sqlserver/connection.py#L421

the conn string expects `SERVER=<host>,<port>` so we need to validate that the port is actually set in the `host` field and if not, check if they added the `port` config

Reply:
we can probably also default to 1433, if its not set in either place && print a debug warning when this happens

Reply:
that is actually handled just below my change, so i opted not to add additional validation.